app/controllers/subscriptions_controller.rb

```diff
@@ -1,5 +1,18 @@
 class SubscriptionsController < ApplicationController
-  before_action :find_rubygem
+  before_action :redirect_to_signin, only: :index, unless: :signed_in?
+  before_action :redirect_to_new_mfa, only: :index, if: :mfa_required_not_yet_enabled?
+  before_action :redirect_to_settings_strong_mfa_required, only: :index, if: :mfa_required_weak_level_enabled?
+
+  before_action :find_rubygem, only: %i[create destroy]
+
+  layout "hammy"
+
+  def index
+    add_breadcrumb t("breadcrumbs.dashboard"), dashboard_path
+    add_breadcrumb t("breadcrumbs.subscriptions")
+
+    @subscribed_gems = current_user.subscribed_gems.with_versions.by_name.preload(:most_recent_version).load_async
```

Comment:
We're loading the same set of data in the `DashboardsController`, could we move this into a scope?

Reply:
I looked at this, and since it's just a basic collection of rubygems with versions (something we do all over and have kind of optimized already), I don't think it's as repetitive as the line length makes it look. The only thing I could merge is the with_versions.by_name and I don't think it would improve the code. I prefer the composability we have now.